packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1370,6 +1381,27 @@ abstract class FlutterCommand extends Command<void> {
     return dartDefineConfigJsonMap;
   }
 
+  String convertEnvFileToJsonRaw(String path, String configRaw) {
+    // convert env file to json
+    final String jsonPairs = configRaw.split('\n')
+    .where((String line) => line.trim().isNotEmpty)
+    .map((String line) {
+      final List<String> parts = line.split('=');
+      if (parts.length != 2) {
+        throwToolExit('.env config define file "--${FlutterOptions
+            .kDartDefineFromFileOption}=$path" format err, '
+            'please fix first! format err:\n$line');
+      }
```

Comment:
I think we need to reconsider this part. Say for example we had a json implementation for dart defines that was the below

```
{
    myVar: "sfadsfv=432f"
}
```

If we wanted to implement that env variable with a `.env` file, we need to allow for more than `=` character. Can you reimplement this so that you are only splitting on the first available matching character and write a test to confirm?

Reply:
Sure, it will be more secure.

I'll do it

Reply:
Great, feel free to ping me on the comments when you have a fix in

Reply:
Ok for me @eliasyishak. I added cases for values containing `=` in JSON and .env

Reply:
I resolve this conversation in order to focus exchanges in one conversation.

We found multiple other one example to manager with @andrewkolos. Feel free to challenge in the related conversation below.